class.jetpack-jitm.php

```diff
@@ -56,490 +55,275 @@ function get_emblem()
-				 'edit-shop_order', // WooCommerce > Orders
-				 'shop_order', // WooCommerce > Edit Order
-			 );
-
-			if ( in_array( $screen->id, $pages_to_display ) ) {
-				add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
-				add_action( 'admin_notices', array( $this, 'woocommerce_services_msg' ) );
-			}
-		}
-	}
-
-	/*
-	 * Present Manage just in time activation msg on update-core.php
-	 *
-	 */
-	function manage_msg() {
-		$normalized_site_url = Jetpack::build_raw_urls( get_home_url() );
-		?>
-		<div class="jp-jitm">
-			<a href="#" data-module="manage" class="dismiss"><span class="genericon genericon-close"></span></a>
-
-			<?php echo self::get_emblem(); ?>
-
-			<p class="msg">
-				<?php esc_html_e( 'Reduce security risks with automated plugin updates.', 'jetpack' ); ?>
-			</p>
-
-			<p>
-				<img class="j-spinner hide" src="<?php echo esc_url( includes_url( 'images/spinner-2x.gif' ) ); ?>" alt="<?php echo esc_attr__( 'Loading...', 'jetpack' ); ?>" /><a href="#" data-module="manage" class="activate button <?php if ( Jetpack::is_module_active( 'manage' ) ) {
-					echo 'hide';
-				} ?>"><?php esc_html_e( 'Activate Now', 'jetpack' ); ?></a><a href="<?php echo esc_url( 'https://wordpress.com/plugins/' . $normalized_site_url ); ?>" target="_blank" title="<?php esc_attr_e( 'Go to WordPress.com to try these features', 'jetpack' ); ?>" id="jetpack-wordpressdotcom" class="button button-jetpack <?php if ( ! Jetpack::is_module_active( 'manage' ) ) {
-					echo 'hide';
-				} ?>"><?php esc_html_e( 'Go to WordPress.com', 'jetpack' ); ?></a>
-			</p>
-		</div>
-		<?php
-		//jitm is being viewed, track it
-		$jetpack = Jetpack::init();
-		$jetpack->stat( 'jitm', 'manage-viewed-' . JETPACK__VERSION );
-		$jetpack->do_stats( 'server_side' );
-	}
-
-	/*
-	 * Present Photon just in time activation msg
-	 *
-	 */
-	function photon_msg() {
-		?>
-		<div class="jp-jitm">
-			<a href="#" data-module="photon" class="dismiss"><span class="genericon genericon-close"></span></a>
-
-			<?php echo self::get_emblem(); ?>
-
-			<p class="msg">
-				<?php esc_html_e( 'Speed up your photos and save bandwidth costs by using a free content delivery network.', 'jetpack' ); ?>
-			</p>
-
-			<p>
-				<img class="j-spinner hide" style="margin-top: 13px;" width="17" height="17" src="<?php echo esc_url( includes_url( 'images/spinner-2x.gif' ) ); ?>" alt="<?php echo esc_attr__( 'Loading...', 'jetpack' ); ?>" /><a href="#" data-module="photon" class="activate button button-jetpack"><?php esc_html_e( 'Activate Photon', 'jetpack' ); ?></a>
-			</p>
-		</div>
-		<?php
-		//jitm is being viewed, track it
-		$jetpack = Jetpack::init();
-		$jetpack->stat( 'jitm', 'photon-viewed-' . JETPACK__VERSION );
-		$jetpack->do_stats( 'server_side' );
+		add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
+		add_action( 'admin_notices', array( $this, 'ajax_message' ) );
+		add_action( 'edit_form_top', array( $this, 'ajax_message' ) );
```

Comment:
Why is this hooking into `admin_notices` and also `edit_form_top`? This produces two XHRs on a post edit screen.